pkg/ccl/changefeedccl/checkpoint/checkpoint.go

```diff
@@ -88,27 +94,35 @@ func Restore(
 	oldCheckpointTs hlc.Timestamp,
 	checkpoint *jobspb.TimestampSpansMap,
 ) error {
+	currHighWater := sf.Frontier()
 	if checkpoint == nil {
-		ts := oldCheckpointTs
-		if ts.IsEmpty() {
```

Comment:
Instead of making this change, it feels to me that the cleaner solution would be to fix ALTER CHANGEFEED to include a timestamp. Alternatively, if upon restore we find that the checkpoint timestamp isn't set, we can set it based on the old assumptions. WDYT?